op-service/sources/l1_beacon_client.go

```diff
@@ -114,17 +211,10 @@ func (cl *L1BeaconClient) GetBlobSidecars(ctx context.Context, ref eth.L1BlockRe
 		return nil, fmt.Errorf("error in converting ref.Time to slot: %w", err)
 	}
 
-	reqPath := path.Join(sidecarsMethodPrefix, strconv.FormatUint(slot, 10))
-	var reqQuery url.Values
-	if !cl.cfg.FetchAllSidecars {
-		reqQuery = url.Values{}
-		for i := range hashes {
-			reqQuery.Add("indices", strconv.FormatUint(hashes[i].Index, 10))
-		}
-	}
-
-	var resp eth.APIGetBlobSidecarsResponse
-	if err := cl.apiReq(ctx, &resp, reqPath, reqQuery); err != nil {
+	f := cl.pool.Get()
+	resp, err := f.BeaconBlobSideCars(ctx, cl.cfg.FetchAllSidecars, slot, hashes)
+	if err != nil {
+		cl.pool.MoveToNext()
 		return nil, fmt.Errorf("failed to fetch blob sidecars for slot %v block %v: %w", slot, ref, err)
 	}
```

Comment:
What do you think about making the fallback mechanism more transparent, ie. trying out the whole pool before returning an error? We could silently fall back and only return an error if all endpoints fail, like the following in its own function
```suggestion
var errs []error
for i := 0; i < cl.pool.Len(); i++ {
	f := cl.pool.Get()
	resp, err := f.BeaconBlobSideCars(ctx, cl.cfg.FetchAllSidecars, slot, hashes)
	if err != nil {
	// log that we're moving to next
		cl.pool.MoveToNext()
		errs = append(errs, fmt.Errorf("failed to fetch blob sidecars for slot %v block %v: %w", slot, ref, err))
	} else {
		return resp, nil
	}
}
return nil, errors.Join(errs)
```

Reply:
this works & is likely a tad faster on errors